src/client/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -263,5 +268,47 @@ public string CreateAuthorizationHeader()
                 _authenticationScheme?.AuthorizationHeaderPrefix ?? TokenType,
                 AccessToken);
         }
+
+#if !NETSTANDARD1_3
+        /// <summary>
+        /// Creates an object to hold the claims from an ID Token.
+        /// </summary>
+        /// <returns>ClaimsPrincipal object containing the claims from the ID Token.</returns>
+        public ClaimsPrincipal GetIdTokenClaims()
+        {
+            if (string.IsNullOrEmpty(IdToken))
```

Comment:
I guess that's the case for client credentials? right?

Reply:
We don't verify the signature of the IDToken (for v2.0 ID Tokens). Maybe we want to indicate it? (not a big problem in practice as it comes from AAD)

Reply:
How can we indicate that we don't verify the signature?

Reply:
@jmprieur @jennyf19 @neha-bhargava 

Today, MSAL parses the IDTOken internally and uses a bunch of claims from it: oid, tid, sub, upn, preferred_username and more. So I think we must have already decided that we don't need to validate the id token?